tor/cmd_onion.go

```diff
@@ -24,26 +33,43 @@ const (
 	V3
 )
 
+const (
+	// V2KeyParam is a parameter that Tor accepts for a new V2 service.
+	V2KeyParam = "RSA1024"
+
+	// V3KeyParam is a parameter that Tor accepts for a new V3 service.
+	V3KeyParam = "ED25519-V3"
+)
+
 // OnionStore is a store containing information about a particular onion
 // service.
 type OnionStore interface {
 	// StorePrivateKey stores the private key according to the
 	// implementation of the OnionStore interface.
-	StorePrivateKey(OnionType, []byte) error
+	StorePrivateKey([]byte) error
 
 	// PrivateKey retrieves a stored private key. If it is not found, then
 	// ErrNoPrivateKey should be returned.
-	PrivateKey(OnionType) ([]byte, error)
+	PrivateKey() ([]byte, error)
 
 	// DeletePrivateKey securely removes the private key from the store.
-	DeletePrivateKey(OnionType) error
+	DeletePrivateKey() error
+}
+
+// EncrypterDecrypter is used for encrypting and decrypting the onion service
+// private key.
+type EncrypterDecrypter interface {
+	EncryptPayloadToWriter(bytes.Buffer, io.Writer) error
```

Comment:
I think instead of `bytes.Buffer` this should just be `[]byte`.

Reply:
Yeah that makes sense as we need to convert it to bytes in the current function implementation anyway. I'll update this here and in the other PR as well